Khan/wonder-blocks
# Fix close icon on modals and update toolbar to match updated Toolbar spec

The updated Toolbar spec aligns the 24x24 bounding box of an IconButton's icon to be 16px in from the left side of the toolbar.  This PR updates the Toolbar examples to remove all pixel nudging that was there previously.  It also updates Modal to put the close icon inside the Toolbar when there is a toolbar instead of having it float above the toolbar.  The close icons on modals without a toolbar are positioned to match the position of the close icon on modals that do have a toolbar.

Test Plan:
- go to the Modal section of the netlify link
- click on some of the close buttons to make sure they're still working for both ModalLauncher examples and the other examples
- see that Modals without the toolbar header have the close buttons with the same top padding as left padding
